source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -803,26 +809,43 @@ def fill_journeys(self, request_type, api_request, instance):
         if not api_request['destination_mode']:
             api_request['destination_mode'] = ['walking']
 
+        # Return the possible couples combinations (origin_mode and destination_mode)
         krakens_call = get_kraken_calls(api_request)
 
+        # min_nb_journeys option
+        if api_request['min_nb_journeys']:
+            min_nb_journeys = api_request['min_nb_journeys']
+        else:
+            min_nb_journeys = api_request['min_nb_journeys'] = 1
+
+        # We need the original request (api_request) for filtering, but request
+        # is modified by create_next_kraken_request function.
         request = deepcopy(api_request)
-        min_asked_journeys = get_or_default(request, 'min_nb_journeys', 1)
+
         min_journeys_calls = get_or_default(request, '_min_journeys_calls', 1)
 
         responses = []
         nb_try = 0
-        nb_qualified_journeys = nb_journeys(responses)
+        nb_qualified_journeys = 0
 
         while request is not None and \
-                ((nb_qualified_journeys < min_asked_journeys and nb_try < min_asked_journeys)
-                 or nb_try < min_journeys_calls):
+                ((nb_qualified_journeys < min_nb_journeys and nb_try < min_nb_journeys) \
```

Comment:
shouldn't ~~nb_try < min_nb_journeys~~ rather be   `nb_try < min_journeys_calls`? (it's not your code though...)

Reply:
+1. Why 'nb_try < min_nb_journeys' ?

Reply:
@mbounabi it's the same as before, just a little refactor on naming

Reply:
You will never send the same request to kraken. Or you break the loop because there is no solution (or only a non-pt solution), or you have a solution and it will change the requested datetime.

Reply:
You don't break the loop if there is no solution, that's my point : it looks like you don't stop until `nb_try == min_nb_journeys`

Reply:
https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/jormungandr/scenarios/new_default.py#L831